File: php/hamming/hamming.php
<?php

function distance($a, $b)
{
  if (strlen($a) !== strlen($b)) {
    throw new InvalidArgumentException('DNA strands must be of equal length.');
  }

  $a_ = str_split($a);
  $b_ = str_split($b);
  $result = 0;

  foreach ($a_ as $k => $v) {
    if ($v !== $b_[$k]) {
      $result += 1;
    }
  }

  return $result;
}

File: go/hamming/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
  if len(a) != len(b) {
    return 0, errors.New("strands must have same length")
  }
  result := 0
  for i := 0; i < len(a); i++ {
    if a[i] != b[i] {
      result++
    }
  }
  return result, nil
}
